firedrake/mesh.py

```diff
@@ -1036,6 +1041,10 @@ def __init__(self, coordinates):
 
         self._coordinates = coordinates
 
+        # Cached coordinate functions in different function spaces
+        # dict: FunctionSpace -> (coordinate Function, interpolator)
+        self.aux_coordinate_functions = {}
```

Comment:
It's unclear to me what is the best way to implement the cache. Here we are just storing a weakref to the Function and Interpolator created elsewhere. One option would be the implement a get method that handles this. But for garbage collection, we can only store weakrefs in `MeshGeometry`.